### PR TITLE
index check when alter or drop tag|edge 

### PR DESCRIPTION
**Pull request by bright-starry-sky**
Refer to #1545.

### CODE REVIEW

File: src/meta/processors/schemaMan/AlterEdgeProcessor.cpp [hunk 1]
```diff
@@ -41,6 +42,50 @@ void AlterEdgeProcessor::process(const cpp2::AlterEdgeReq& req) {
 
     // Update schema column
     auto& edgeItems = req.get_edge_items();
+
+    // Check the edge belongs to the index
```
Comment (Amber1990Zhang) [inline src/meta/processors/schemaMan/AlterEdgeProcessor.cpp]: check if the edge contains indexes

File: src/meta/processors/schemaMan/AlterTagProcessor.cpp [hunk 2]
```diff
@@ -41,6 +43,50 @@ void AlterTagProcessor::process(const cpp2::AlterTagReq& req) {
 
     // Update schema column
     auto& tagItems = req.get_tag_items();
+
+    // Check the tag belongs to the index
+    {
```
Comment (dangleptr) [inline src/meta/processors/schemaMan/AlterTagProcessor.cpp]: Could we unify the code

**Comment by bright-starry-sky** [comment]
Addressed comments.

File: src/meta/processors/schemaMan/AlterEdgeProcessor.cpp [hunk 3]
```diff
@@ -56,7 +75,7 @@ void AlterEdgeProcessor::process(const cpp2::AlterEdgeReq& req) {
 
     // Update schema property
     auto& alterSchemaProp = req.get_schema_prop();
-    auto retCode = MetaServiceUtils::alterSchemaProp(columns, prop, std::move(alterSchemaProp));
```
Comment (darionyaphet) [inline src/meta/processors/schemaMan/AlterEdgeProcessor.cpp]: why remove `move`?

File: src/meta/processors/BaseProcessor.inl [hunk 4]
```diff
@@ -432,5 +432,58 @@ void BaseProcessor<RESP>::doSyncMultiRemoveAndUpdate(std::vector<std::string> ke
...
+
+template<typename RESP>
+cpp2::ErrorCode
+BaseProcessor<RESP>::indexCheck(const std::vector<nebula::cpp2::IndexItem>& items,
+                                const std::vector<cpp2::AlterSchemaItem>& alterItems) {
+    for (const auto& index : items) {
+        for (const auto& tagItem : alterItems) {
+            if (tagItem.op == nebula::meta::cpp2::AlterSchemaOp::CHANGE ||
+                tagItem.op == nebula::meta::cpp2::AlterSchemaOp::DROP) {
+                const auto& tagCols = tagItem.get_schema().get_columns();
```
Comment (critical27) [inline src/meta/processors/BaseProcessor.inl]: It seems `columns` is better, but not a big deal.

File: src/meta/processors/schemaMan/AlterEdgeProcessor.cpp (hunk 3 above)
Comment (bright-starry-sky) [reply src/meta/processors/schemaMan/AlterEdgeProcessor.cpp]: req.get_schema_prop() return a const object , so there is no need for std::move

**Comment by jude-zhu** [comment]
close #462

**Pull request merged**